### DISCUSSION TITLE
Maybe better to create a sub class for valid class declaration?

### DISCUSSION THREAD

**Discussion by williamfzc** [tiangolo/pydantic-sqlalchemy#94] [Questions] [answered]
hi there :)

```python
PydanticUser = sqlalchemy_to_pydantic(User)

class PydanticUser2(sqlalchemy_to_pydantic(User)):
    pass

print(PydanticUser, PydanticUser2, User)
```

the output is:

```python
<class 'User'> <class '__main__.PydanticUser2'> <class '__main__.User'>
```

If we directly use `sqlalchemy_to_pydantic`, actually we create another `User` class in namespace, which may be confused.

**Comment by ljluestc** [comment]
from pydantic import BaseModel
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy import Column, Integer, String

Base = declarative_base()

class User(Base):
    __tablename__ = 'users'
    id = Column(Integer, primary_key=True, index=True)
    username = Column(String, unique=True, index=True)
    email = Column(String, unique=True, index=True)

class BasePydanticModel(BaseModel):
    class Config:
        orm_mode = True

PydanticUser = sqlalchemy_to_pydantic(User, config=BasePydanticModel)
class PydanticUser2(BasePydanticModel, PydanticUser):
    pass

print(PydanticUser, PydanticUser2, User)

**Comment by tiangolo** [comment, marked as answer]
Thanks for the patience with my reply! :sweat_smile:

I deprecated this library a while ago in favor of SQLModel which would solve all the same use cases in a much better way, and I'm directing all the effort in that direction.

SQLModel uses Pydantic models to define both data and table (SQL) models, so it's easier to do all these things. :rocket:
